src/Altinn.App.Core/Helpers/DataModel/FormDataWrapperFactory.cs

```
@@ -0,0 +1,55 @@
+using System.Collections.Frozen;
+using System.Diagnostics;
+using Altinn.App.Core.Features;
+
+namespace Altinn.App.Core.Helpers.DataModel;
+
+internal static class FormDataWrapperFactory
+{
+    private static readonly FrozenDictionary<Type, Type> _pathAccessors = InitializePathAccessorLookup();
+
+    private static FrozenDictionary<Type, Type> InitializePathAccessorLookup()
+    {
+        var interfaceType = typeof(IFormDataWrapper<>);
+        return AppDomain
+            .CurrentDomain.GetAssemblies()
+            .SelectMany(a =>
+            {
+                List<KeyValuePair<Type, Type>> pathAccessors = new();
+                try
+                {
+                    var assemblyTypes = a.GetExportedTypes();
+                    foreach (var type in assemblyTypes)
+                    {
+                        var formDataWrapperInterface = type.GetInterfaces()
+                            .FirstOrDefault(i => i.IsGenericType && i.GetGenericTypeDefinition() == interfaceType);
+                        if (formDataWrapperInterface is not null)
+                        {
+                            pathAccessors.Add(
+                                KeyValuePair.Create(formDataWrapperInterface.GenericTypeArguments[0], type)
+                            );
+                        }
+                    }
+                }
+                catch (Exception)
+                {
+                    // Just ignore it if we can't load the assembly types for some reason
+                    // It is likely not the relevant assembly anyway
```

Completely empty catch clauses are kind of scary. What if we make a bug in the try block? Can you think of any good ways to let us disover potential bugs here (aside from the assembly thing)? E.g.

* Make this type non-static, inject logger and emit warning?
* Debug assertion so that we are made aware during local testing?

---

The real fault situation here is when we have to return `ReflectionFormDataWrapper` instead of a source generated one, so that's probably the correct time to log a warning. 

I rewrote this part to let all the generated classes have a static `[ModuleInitializer]` and register themself instead of relying on assembly scanning, but in reality it did not change the possible failure of not finding a generated wrapper (apart from removing the need to catch exceptions)

---

I was more referring to if we introduced bug in the old try block, it would be hard to find out ;) But yes I agree it would be nice to have some logging for the reflection vs src gen thing as well. Perhaps during startup, but I'm fine with leaving that for later. 

The new solution looks fine to me, a couple of drawbacks:
* Slower lookups during runtime
* Exposes a public function that isn't really meant to be called by any other codepath

---

I don't really see why this would cause slower lookups during runtime.

I use `Dictionary<Type, (object)=>IFormDataValidator>` instead of `FrozenDictionary<Type,Type>` followed by `Activator.CreateInstance`. Do you think the frozen dictionary should be obviously faster?

The public function is not ideal, but not something I think we should worry about.